src/csm_testing/tests/iuf_stages/__main__.py

```diff
@@ -0,0 +1,484 @@
+        resp = session.get(dummy_repo_url, verify=ca_cert_path, auth=auth)
+
+        if resp.status_code == 200:
+            print(f"TEST CASE: Repository '{dummy_repo_url}' exists.")
+            TEST_CASES += 1
+        elif resp.status_code == 404:
+            print(f"ERROR: Repository '{dummy_repo_url}' not found.")
+            print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+            sys.exit(1)
+        else:
+            print(f"ERROR: Failed to fetch repository: {resp.status_code}, {resp.text}")
+            print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+            sys.exit(1)
+
+        repo_info = resp.json()
+        if repo_info:
+            print(f"INFO: Repository Name: {repo_info['name']}")
+            print(f"INFO: Default Branch: {repo_info['default_branch']}")
+    except subprocess.CalledProcessError as err:
+        print(f"Error: {err}")
+        print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+        sys.exit(1)
+
+def update_cfs_config(activity_name): # pylint: disable=missing-function-docstring
+    global TEST_CASES  # pylint: disable=global-statement
+    try:
+        command = (
+            f"iuf -a {activity_name} -m {MEDIA_DIR} run -rv {MEDIA_DIR}/product_vars.yaml "
+            f"-bm {MEDIA_DIR}/management-bootprep.yaml -r update-cfs-config"
+        )
+        run_command(command)
+
+        kubectl_command = (
+            f"kubectl get configmaps -n argo {activity_name} -o jsonpath="
+            "'{.data.iuf_activity}' | jq '.operation_outputs.stage_params"
+            '["update-cfs-config"]["update-management-cfs-config"]'
+            '["sat-bootprep-run"].script_stdout' + "' | xargs -0 echo -e"
+        )
+
+        kubectl_output, _ = run_command(kubectl_command)
+
+        if not kubectl_output:
+            print(
+                f"ERROR: The output for cfs configurations is null or empty in the "
+                f"{activity_name} configmap."
+            )
+            print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+            sys.exit(1)
+        else:
+            print("TEST CASE: Configmap updated with cfs configurations data")
+            TEST_CASES += 1
+
+        cfs_command = "cray cfs configurations list | grep 'config-minimal-management-dummy-1.0.0'"
+        cfs_output, _ = run_command(cfs_command)
+        if not cfs_output:
+            print("ERROR: No configuration found in cfs")
+            print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+            sys.exit(1)
+        print(f"TEST CASE: CFS configuration found {cfs_output}")
+        TEST_CASES += 1
+    except subprocess.CalledProcessError as err:
+        print(f"Error: {err}")
+        print(f"INFO: Total test cases executed for stage operations: {TEST_CASES}")
+        sys.exit(1)
+
+def prepare_images(activity_name): # pylint: disable=too-many-locals
```

Comment:
If pylint is raising the `too-many-locals` error, it is a strong indication that the function in question should be refactored, possibly splitting it into multiple functions. This is one of the more important pylint errors to actually address, because it can make code much harder to understand and maintain.

Reply:
I split the function in 2 parts. Also I split iuf_stages into 2 files - __main__.py and helper.py
helper.py will have all the helper functions.